src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -214,6 +245,24 @@ public Connection getDbInstance() throws SQLException {
         return DriverManager.getConnection(url);
     }
 
+    /**
+     * This method closes database connection.
+     * @throws SQLException when fails to close database connection.
+     */
+    synchronized void close() throws SQLException {
+        try (LockScope ignored = LockScope.lock(dbConnectionLock.writeLock())) {
```

Comment:
in some cases this will upgrade from a read to write lock (if corruption during getAllSpoolMessageIds, etc), which will fail